Editorial suggestions (#192)
ietf-wg-masque/draft-ietf-masque-h3-datagram: draft-ietf-masque-h3-datagram.md

{{?RFC6455}}.

In {{datagrams}}, this document describes HTTP Datagrams, a convention that

> supports the bidirectional and possibly multiplexed exchange of data inside an

Author: I find the use of "possibly" to be awkward. The protocol supports the feature, it doesn't possibly support it.

> supports bidirectional and optionally multiplexed data exchange inside an

Reviewer: I like optionally but prefer the rest of the previous wording
```suggestion
supports the bidirectional and optionally multiplexed exchange of data inside an
```

### The SETTINGS_H3_DATAGRAM HTTP/3 Setting {#setting}

> Implementations of HTTP/3 that are willing to receive HTTP Datagrams can

Author: Generally we're using "HTTP Datagrams" to mean things that work even with capsules, and "HTTP/3 Datagrams" to mean the things that go in QUIC datagrams. So this seemed incorrect.

> Datagrams  by sending the SETTINGS_H3_DATAGRAM setting with a

Author: ```suggestion
Datagrams by sending the SETTINGS_H3_DATAGRAM setting with a
```

> Implementations that are willing to receive HTTP/3 Datagrams can indicate
> support to their peer by sending the SETTINGS_H3_DATAGRAM setting with a

Reviewer: We specifically removed the term support since that refers to the implementation not the intent
```suggestion
Endpoints can indicate to their peer that they are willing to receive HTTP/3
Datagrams  by sending the SETTINGS_H3_DATAGRAM setting with a
```

with error H3_SETTINGS_ERROR. In all cases, the maximum permitted value of the
SETTINGS_H3_DATAGRAM setting parameter is 1.

It is RECOMMENDED that implementations that support receiving HTTP/3 Datagrams

> using QUIC always send the SETTINGS_H3_DATAGRAM setting with a value of 1,

Reviewer: ```suggestion
always send the SETTINGS_H3_DATAGRAM setting with a value of 1,
```

## HTTP Datagrams using Capsules

When HTTP/3 Datagrams are unavailable or undesirable, HTTP Datagrams can be sent

> using the Capsule Protocol, see {{datagram-capsule}}. Capsules can also be used

Author: Adding a note to explain how capsules will be used in things like CONNECT-IP, may partly address #159.

Reviewer: I like this note, but it doesn't make sense in the "HTTP Datagrams using Capsules" section. Can we move it the Capsule Protocol section?

Author: Moved down

> to communicate reliable and bidirectional control messages associated with a

Reviewer: They're not necessarily associated with a datagram-based protocol

Author: I moved this down, but kept the language. The point is to clarify that although you're pointing out that you use it when you can't use QUIC datagrams, you still will want capsules for your datagram-based protocol. If you have other text in mind, please add it.

Reviewer: Fair enough, this works for me

support QUIC DATAGRAM frames, it might not need a stream encoding. Additionally,
HTTP extensions can use HTTP Datagrams with their own data stream protocol.
However, new HTTP extensions that wish to use HTTP Datagrams SHOULD use the

> Capsule Protocol unless they have a good reason not to.

Author: "...good reason not to" sounds weak.

Reviewer: SGTM
